### PR TITLE
Fix minor issue in IfToSwitch refactoring

### PR DESCRIPTION
**Pull request by alrz**

### CODE REVIEW

File: src/Features/Core/Portable/UseSystemHashCode/UseSystemHashCodeCodeFixProvider.cs [hunk 1]
```diff
@@ -43,7 +43,7 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
             SyntaxEditor editor, CancellationToken cancellationToken)
         {
             var generator = SyntaxGenerator.GetGenerator(document);
-            var declarationService = document.GetLanguageService<ISymbolDeclarationService>();
+            var declarationService = document.GetLanguageService<ISymbolDeclarationService>()!;
```
Comment (jasonmalinowski) [inline src/Features/Core/Portable/UseSystemHashCode/UseSystemHashCodeCodeFixProvider.cs]: You can do document.Project.LanguageServices.GetRequiredService() which returns non-nullable (and throws if ever this is somehow broken.)

(Although I'm not understanding why we're seeing this in this PR -- is GitHub acting funny?)

File: src/Features/Core/Portable/ConvertIfToSwitch/AbstractConvertIfToSwitchCodeRefactoringProvider.cs [hunk 2]
```diff
@@ -45,7 +45,7 @@ public sealed override async Task ComputeRefactoringsAsync(CodeRefactoringContex
 
             var operations = parentBlock.Operations;
             var index = operations.IndexOf(ifOperation);
-            var analyzer = CreateAnalyzer(document.GetLanguageService<ISyntaxFactsService>(), ifStatement.SyntaxTree.Options);
+            var analyzer = CreateAnalyzer(document.GetLanguageService<ISyntaxFactsService>()!, ifStatement.SyntaxTree.Options);
```
Comment (jasonmalinowski) [inline src/Features/Core/Portable/ConvertIfToSwitch/AbstractConvertIfToSwitchCodeRefactoringProvider.cs]: You can do document.Project.LanguageServices.GetRequiredService() which returns non-nullable (and throws if ever this is somehow broken.)

Comment (dibarbet) [reply src/Features/Core/Portable/ConvertIfToSwitch/AbstractConvertIfToSwitchCodeRefactoringProvider.cs]: @alrz you probably will need to update from the latest master.  I fixed this last night when doing infra work, CI was broken
https://github.com/dotnet/roslyn/blob/master/src/Features/Core/Portable/ConvertIfToSwitch/AbstractConvertIfToSwitchCodeRefactoringProvider.cs#L48

If you want to change it to getrequired as Jason suggested I wouldn't mind :)

File: src/Features/Core/Portable/UseSystemHashCode/UseSystemHashCodeCodeFixProvider.cs (hunk 1 above)
Comment (dibarbet) [reply src/Features/Core/Portable/UseSystemHashCode/UseSystemHashCodeCodeFixProvider.cs]: Same comment as above.

Comment (alrz) [reply src/Features/Core/Portable/UseSystemHashCode/UseSystemHashCodeCodeFixProvider.cs]: looks like a bad merge?

**Comment by alrz** [comment]
> which bug is this fixing

See `TestMissingOnSubsequentBlock`. A fix was incorrectly offered in this scenario which produces bad code.

**Comment by alrz** [comment]
Rebased.

**Comment by jasonmalinowski** [comment]
I guess I was a bit confused by this comment that's before ParseIfStatement:

https://github.com/dotnet/roslyn/pull/39465/files#diff-568c6a3bb34fe548f92d1d3ee7d648d5R132-R136

I had read the last line in the grammar as implying that we do handle nested blocks of if statements, hence it was confusing to me the test was testing a lack of fix. And to be clear I don't think it's worth us trying to support it, but just trying to reconcile understanding.

**Comment by alrz** [comment]
Adjusted comment and code to the new behavior.

**Comment by jasonmalinowski** [comment]
As usual thank you @alrz for the fix and also all the great tests!

**Pull request merged**